devops/azure-pipelines/azure-pipelines.yml

```diff
@@ -37,20 +40,26 @@ stages:
           dockerPath: '**/mavenpipe.dockerfile'
 
   - stage: Deploy
+    condition: >
+      and( succeeded(), 
+      in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'), 
```

Comment:
usually we like just about every kind of build besides PullRequest (e.g. Manual) to deploy, but up to you.